velox/vector/FlatVector-inl.h

```diff
@@ -453,7 +453,10 @@ void FlatVector<T>::ensureWritable(const SelectivityVector& rows) {
 
     if constexpr (std::is_same_v<T, bool>) {
       auto rawNewValues = newValues->asMutable<uint64_t>();
-      std::memcpy(rawNewValues, rawValues_, values_->size());
+      std::memcpy(
+          rawNewValues,
+          rawValues_,
+          std::min(values_->size(), newValues->size()));
     } else {
```

Comment:
@mbasmanova It seems that Line 451-452 can be moved into this block. Is my understanding correct?

Reply:
Yes, I think so.